core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -234,6 +237,19 @@ open class TileInfo : IsPartOfGameInfoSerialization {
         return player.viewableTiles.contains(this)
     }
 
+    fun isExplored(player: CivilizationInfo): Boolean {
+        if (UncivGame.Current.viewEntireMapForDebug || player.isSpectator())
+            return true
+        return exploredBy.contains(player.civName) || player.exploredTiles.contains(position)
+    }
+
+    fun setExplored(player: CivilizationInfo, isExplored: Boolean) {
```

Comment:
Here too we need to save the explored tiles the old way, so that another player doesn't override them for me 
In multiplayer, each player gets the current game and returns the new serialized version. Players on old versions will not even serialize fields that their version doesn't contain!